Rubberduck.Parsing/VBA/DeclarationCaching/DeclarationFinder.cs

```diff
@@ -1338,13 +1338,23 @@ public IEnumerable<Declaration> FindNewDeclarationNameConflicts(string newName,
             identifierMatches = identifierMatches.Where(nc => !IsEnumOrUDTMemberDeclaration(nc)).ToList();
             var referenceConflicts = identifierMatches.Where(idm =>
                 renameTarget.References
-                    .Any(renameTargetRef => renameTargetRef.ParentScoping == idm.ParentDeclaration
+                    .Any(renameTargetRef => 
+                        renameTargetRef.ParentScoping == idm.ParentDeclaration
+                        
                         || !renameTarget.ParentDeclaration.DeclarationType.HasFlag(DeclarationType.ClassModule)
                             && idm == renameTargetRef.ParentScoping
                             && !UsesScopeResolution(renameTargetRef.Context.Parent)
+
                         || idm.References
                             .Any(idmRef => idmRef.ParentScoping == renameTargetRef.ParentScoping
-                                && !UsesScopeResolution(renameTargetRef.Context.Parent)))
+                                && renameTargetRef.QualifiedModuleName != renameTarget.QualifiedModuleName
+                                && !UsesScopeResolution(renameTargetRef.Context.Parent))
+
+                        || idm.References
+                            .Any(idmRef => idmRef.ParentScoping == renameTargetRef.ParentScoping
+                                && renameTargetRef.QualifiedModuleName == renameTarget.QualifiedModuleName
+                                && !UsesScopeResolution(idmRef.Context.Parent)))
+
```

Comment:
Had to come back here to say that's a very nice & welcome readability boost right there, good job! 👍